postgresql/resource_postgresql_grant.go

```diff
@@ -289,7 +295,12 @@ WHERE grantee = $2
 		return fmt.Errorf("could not read privileges for schema %s: %w", dbName, err)
 	}
 
-	d.Set("privileges", pgArrayToSet(privileges))
+	granted := pgArrayToSet(privileges)
+	wanted := d.Get("privileges").(*schema.Set)
+	equal := arePrivilegesEqual(granted, wanted, d)
+	if !equal {
+		return d.Set("privileges", wanted)
+	}
```

Comment:
In addition to the comment above about `wanted`/`granted`, for all these calls, the `equal` variable could be removed, and if we take into account my first comment about the function argument, it could look like:

```suggestion
	if !resourcePrivilegesEqual(granted, d)
		return d.Set("privileges", granted)
	}
```

Reply:
inlined the if statement with 8dfcf905d1bfb5746001a6e98fca82492058bb1b